soroban-sdk/src/env.rs

```diff
@@ -127,28 +111,28 @@ impl Env {
         Ledger::new(self)
     }
 
-    /// Get an [Accounts] for accessing accounts in the current ledger.
-    #[inline(always)]
-    pub fn accounts(&self) -> Accounts {
-        Accounts::new(self)
-    }
-
     /// Get a deployer for deploying contracts.
     #[inline(always)]
     pub fn deployer(&self) -> Deployer {
         Deployer::new(self)
     }
 
-    /// Get the 32-byte hash identifier of the current executing contract.
-    pub fn current_contract(&self) -> BytesN<32> {
-        let id = internal::Env::get_current_contract(self);
-        unsafe { BytesN::<32>::unchecked_new(id.in_env(self)) }
+    pub fn current_contract_account(&self) -> Account {
```

Comment:
What does it mean to be able to access the "current contract account". Do all contracts have accounts?

Reply:
Yeah, this is an `Account` object corresponding to the currently running contract. As all the invocations now require `Account` there needs a way to do something on behalf of the currently running contract. This is functionally equivalent to the contract invoker now, but unlike invoker it's extensible (e.g. it's easy to allow contract to safely authorize subcontract calls if needed).